bjs2-70450

SAM19991 opened the pull request.

SAM19991 committed:
bjs2-70450

iBerbasov commented on src/main/java/school/faang/bjs270450/Email.java:
```diff
@@ -0,0 +1,18 @@
+    private String body;
+    public boolean isImportant;
+
+    public Email(String subject, String body, boolean isImportant) {
```
Подобные конструкторы без дополнительной логики можно отдавать так же Ломбоку

iBerbasov commented on src/main/java/school/faang/bjs270450/Main.java:
```diff
@@ -0,0 +1,31 @@
+        );
+
+        Predicate<Email> importantFilter = Email::isImportant;
+
```
Много лишних пустых строк немного затрудняют чтение. 
Обычно подобные тесты оформляют в стиле ААА (act arrange assert), где пустыми строками разделяются три этих основных блока

iBerbasov commented on src/main/java/school/faang/bjs270450/EmailProcessor.java:
```diff
@@ -0,0 +1,19 @@
+public class EmailProcessor {
+
+    public void processEmails(List<Email> emails, Predicate<Email> filter, Function<Email, String> transform, Consumer<Email> handler) {
+        for (Email email : emails) {
```
Это можно переписать поинтереснее с использованием Stream API
На мой субъективный взгляд выглядит красиво и понятно

Но для этого конкретного примера ниже, придется слегка отклониться от примера в условии задачи и изменить transform
```
emails.stream()
      .filter(filter)
      .map(transform)
      .forEach(handler);
```

iBerbasov commented on src/main/java/school/faang/bjs270450/EmailProcessor.java:
```diff
@@ -0,0 +1,19 @@
+
+public class EmailProcessor {
+
+    public void processEmails(List<Email> emails, Predicate<Email> filter, Function<Email, String> transform, Consumer<Email> handler) {
```
Когда аргументов много и они подобного трудночитаемого вида, лучше распологать их на новой строке, так проще читать и сразу видно, что приходит в метод
```
public void processEmails(List<Email> emails, 
                          Predicate<Email> filter, 
                          Function<Email, String> transform, 
                          Consumer<Email> handler) { 
        //...
}
```

SAM19991 committed:
Removed extra lines
Added @Data

SAM19991 committed:
Removed extra lines
Added @Data
Replaced Lambda with Stream
Changed the location of parameters in the method
